### DISCUSSION TITLE
Any way to get total amount of views for a URL through the API?

### DISCUSSION THREAD

**Discussion by Fawesum** [umami-software/umami#659] [Q&A]
Trying to get the total number for a set URL, but it seems I have to get the max date range, then find the URL in the list? Any way to send a "pageviews WHERE url = x" kind of query through the API?

**Comment by Muphet** [comment]
i bet more people will find this article, well, just like me.

docs for API are incomplete and don't mention all parameters but you can indeed pass url, referrer, os, browser, device or country to your query.

**Comment by nergmada** [comment]
Can confirm if you send a request such as `/api/website/{your-website-id}/pageviews?url=/somepage` this will return page views. 

I'm currently trying to work out if you can do advanced querying because right now, pages get recorded with query params e.g. `'/somepage?xyz=abc' !== '/somepage'`. Though, the fact that querystrings are not pruned or are treated as distinct may be a bug

**Reply by nergmada** [reply]
Follow up to this, yes you can! It's a bit hacky because it leans into the Prisma ODM that sits under the hood of Umami but here's how I did it and here's what you can use as reference.

Say your Umami records the following pages:
page | views
-------|---------
/alpha | 4
/alpha?abc | 2

You could run a query such as:

`/api/website/{you-website-id}/pageviews?url=/alpha`

Which would return 4 (technically not, because obvs the output data is split by the unit you specify etc. But we're just assuming a simplified result for now).

If however, you want the system to aggregate it for you, you can use a library such as `qs`, and parse a JSON object of the form:

```json
{
  "url": {
    "startsWith": "/alpha"
  }
}
```
This would result in a querystring as follows:

`?url[startsWith]=/alpha`

Your request, with this new query string should now return 6, (the sum of the 4 from `/alpha` and the 2 from `/alpha?abc`).

You can therefore build complex queries using Prisma's extended filtering syntax, found here: https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting
